dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -375,23 +389,45 @@ private static ArtifactResults generateArtifactReport(
     return upperBoundFailures;
   }
 
+  /**
+   * Partitions {@code symbolProblems} by the JAR file that contains the {@link ClassFile}.
+   *
+   * <p>For example, {@code classes = result.get(JarX).get(SymbolProblemY)} where {@code classes}
+   * are not null means that {@code JarX} has {@code SymbolProblemY} and that {@code JarX} contains
+   * {@code classes} which reference {@code SymbolProblemY.getSymbol()}.
+   */
+  private static ImmutableMap<Path, ImmutableSetMultimap<SymbolProblem, String>> indexByJar(
+      ImmutableSetMultimap<SymbolProblem, ClassFile> symbolProblems) {
+
+    ImmutableMap<Path, Collection<Entry<SymbolProblem, ClassFile>>> jarMap =
+        Multimaps.index(symbolProblems.entries(), entry -> entry.getValue().getJar()).asMap();
+
+    return ImmutableMap.copyOf(
+        Maps.transformValues(
+            jarMap,
+            entries ->
+                ImmutableSetMultimap.copyOf(
+                    Multimaps.transformValues(
+                        ImmutableSetMultimap.copyOf(entries), ClassFile::getClassName))));
+  }
+
   @VisibleForTesting
   static void generateDashboard(
       Configuration configuration,
       Path output,
       List<ArtifactResults> table,
       List<DependencyGraph> globalDependencies,
-      LinkageCheckReport linkageCheckReport,
+      ImmutableMap<Path, ImmutableSetMultimap<SymbolProblem, String>> symbolProblemTable,
       ListMultimap<Path, DependencyPath> jarToDependencyPaths)
       throws IOException, TemplateException {
-    
+
     Map<String, String> latestArtifacts = collectLatestVersions(globalDependencies);
 
     Map<String, Object> templateData = new HashMap<>();
     templateData.put("table", table);
     templateData.put("lastUpdated", LocalDateTime.now());
     templateData.put("latestArtifacts", latestArtifacts);
-    templateData.put("jarLinkageReports", linkageCheckReport.getJarLinkageReports());
+    templateData.put("jarToSymbolProblemToClasses", symbolProblemTable);
```

Comment:
The name jarToSymbolProblemToClasses is getting unwieldy. I'm not sure I have a better idea though. Maybe just symbolProblems.

Reply:
Updated to symbolProblems. (I thought [Hangarian notation](https://en.wikipedia.org/wiki/Hungarian_notation) is easy to understand in Freemarker template, but `jarToSymbolProblemToClasses` is too much)